src/pool/srv_iv.c

```diff
@@ -302,7 +302,7 @@ pool_iv_conn_lookup(struct pool_iv_conns *conns, uuid_t uuid)
 	struct pool_iv_conn	*conn = conns->pic_conns;
 	char			*end = (char *)conn + conns->pic_size;
 
-	while ((char *)conn < end) {
+	while (conn->pic_creds <= end) {
```

Comment:
What is the problem with the original code?

Reply:
In my test I ever got a segfault in pool_iv_conn_lookup() as described in DAOS-5654 (not reproducible every time).
By checking code, it seems to me that  original checking conn valid by "conn->pic_creds <= end" looks not enough, for example a few garbage or dirty bytes left in previous pool_iv_conns but no enough space for one whole iv_conn?

Reply:
It feels like the real issue is that conns isn't properly set up. If that issue is fixed, the original checks here should work fine.

Reply:
i suspected it too, but did not find other errors when insert/delete.
on the other hand, if the pic_cred_size possible difference for different pool conn, it seems possible leave garbage data when different pools connect/disconnect, this is what this patch want to refine.

Reply:
OK... Thanks for elaborating.